modules/tensor_mechanics/src/auxkernels/RankTwoScalarAux.C

```diff
@@ -52,8 +53,34 @@ RankTwoScalarAux::computeValue()
    case 6:
      val = calcEigenValues();
      break;
+   case 7: // VolumetricStrain (Lagrangian)
+     val = _tensor[_qp].trace() +
+           _tensor[_qp](x,x)*_tensor[_qp](y,y) +
+           _tensor[_qp](y,y)*_tensor[_qp](z,z) +
+           _tensor[_qp](z,z)*_tensor[_qp](x,x) +
+           _tensor[_qp](x,x)*_tensor[_qp](y,y)*_tensor[_qp](z,z);
+     break;
+   case 8: // FirstInvariant
+     val = _tensor[_qp].trace();
+     break;
+   case 9: // SecondInvariant
```

Comment:
Perhaps this is the [sum of principal minors form](https://en.wikipedia.org/wiki/Invariants_of_tensors)?